static/js/components/ErrorMessage_test.js

```diff
@@ -0,0 +1,214 @@
+      messageAttributes.forEach(([msgAttribute, message]) => {
+        it(`should render an error and message on the ${codeAttribute} and ${msgAttribute} attributes`, () => {
+          let that = {
+            errorInfo: {
+              [codeAttribute]: code,
+              [msgAttribute]: message,
+            }
+          };
+          let errorMessage = renderErrorMessage(that);
+          assert.equal(true, errorMessage.includes(String(code)));
+          assert.equal(true, errorMessage.includes(errorString));
+          assert.equal(true, errorMessage.includes(contactExpectation));
+          assert.equal(true, errorMessage.includes(message));
+        });
+      });
+    });
+  });
+
+  describe('showing errors on pages', () => {
+    let renderComponent, helper, patchUserProfileStub, listenForActions;
+    let dashboardErrorActions;
+
+    const confirmErrorMessage = (div, expectedMessageText) => {
+      let alert = div.querySelector('.alert-message');
+      let messages = alert.getElementsByTagName('p');
+      assert.deepEqual(messages[0].textContent, expectedMessageText[0]);
+      assert.deepEqual(messages[1].textContent, expectedMessageText[1]);
+      assert.deepEqual(messages[2].textContent, contactExpectation);
+    };
+
+    beforeEach(() => {
+      helper = new IntegrationTestHelper();
+      renderComponent = helper.renderComponent.bind(helper);
+      patchUserProfileStub = helper.sandbox.stub(api, 'patchUserProfile');
+      listenForActions = helper.listenForActions.bind(helper);
+
+      dashboardErrorActions = [
+        RECEIVE_DASHBOARD_FAILURE,
+        RECEIVE_GET_USER_PROFILE_SUCCESS
+      ];
+
+      helper.profileGetStub.
+        withArgs(SETTINGS.username).
+        returns(
+          Promise.resolve(Object.assign({}, USER_PROFILE_RESPONSE, {
+            username: SETTINGS.username
+          }))
+        );
+    });
+
+    afterEach(() => {
+      helper.cleanup();
+    });
+
+    describe('dashboard page', () => {
+      it('error from the backend triggers error message in dashboard', () => {
+        helper.dashboardStub.returns(Promise.reject(ERROR_RESPONSE));
+
+        return renderComponent("/dashboard", dashboardErrorActions, false).then(([, div]) => {
+          confirmErrorMessage(div, [`AB123 ${errorString}`, 'Additional info: custom error message for the user.']);
+        });
+      });
+
+      it('the error from the backend does not need to be complete', () => {
+        let response = _.cloneDeep(ERROR_RESPONSE);
+        delete response.user_message;
+        helper.dashboardStub.returns(Promise.reject(response));
+
+        return renderComponent("/dashboard", dashboardErrorActions, false).then(([, div]) => {
+          confirmErrorMessage(div, [`AB123 ${errorString}`, '']);
+        });
+      });
+
+      it('the error from the backend does not need to exist at all as long as there is an http error', () => {
+        helper.dashboardStub.returns(Promise.reject({}));
```

Comment:
ok, this test was just copypasta from what we had in `DashboardPage_test.js`. I'll add a status code.